lib/ASTGen/Sources/ASTGen/Types.swift

```diff
@@ -380,3 +380,104 @@ extension ASTGenVisitor {
     }.bridgedArray(in: self)
   }
 }
+
+extension ASTGenVisitor {
+  struct GeneratedGenericArguments {
+    var arguments: BridgedArrayRef = .init()
+    var range: BridgedSourceRange = .init()
+  }
+
+  /// Generate 'TypeRepr' from a expression, because 'conformances' arguments in
+  /// macro role attributes are parsed as normal expressions.
+  func generateTypeRepr(
+    expr: ExprSyntax,
+    genericArgs: GeneratedGenericArguments = GeneratedGenericArguments()
+  ) -> BridgedTypeRepr? {
+    switch expr.as(ExprSyntaxEnum.self) {
+
+    case .typeExpr(let node):
+      guard genericArgs.arguments.isEmpty else {
+        break
+      }
```

Comment:
Rather than checking for every node that isn't a decl-ref or member, would it be better to do this check at the top of the function? Could also remove the `node.expression.is(MemberAccessExprSyntax.self) || node.expression.is(DeclReferenceExprSyntax.self)` check from `case .genericSpecializationExpr`

Reply:
Nice :+1: